core/objects.go

```diff
@@ -347,6 +348,13 @@ func (ch Challenge) checkConsistency() error {
 	return nil
 }
 
+func (ch Challenge) GenerateID() string {
```

Comment:
Let's add a comment here so nobody mistakes this for a security mechanism. Explain that this is strictly to help ensure clients obey the spec by preventing them from hardcoding a specific path for specific challenge types.